src/core/mac/sub_mac.cpp

```diff
@@ -328,6 +352,14 @@ Error SubMac::Send(void)
         break;
     }
 
+#if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
+    if (mRadioFilterEnabled)
+    {
+        mCallbacks.TransmitDone(mTransmitFrame, nullptr, mTransmitFrame.GetAckRequest() ? kErrorNoAck : kErrorNone);
```

Comment:
Post `TransmitDone` in a Task to avoid calling the callback within the `Send` ?

Reply:
There is no need for this.  The callback can be called from `Send()` fine (already happens if tx is aborted).